core/src/main/java/org/jruby/javasupport/ext/JavaLang.java

```diff
@@ -59,32 +59,28 @@
 public abstract class JavaLang {
 
     public static void define(final Ruby runtime) {
-        Iterable.define(runtime);
-        Comparable.define(runtime);
-        Throwable.define(runtime);
-        Runnable.define(runtime);
-        Character.define(runtime);
-        Number.define(runtime);
-        Class.define(runtime);
-        ClassLoader.define(runtime);
+        JavaExtensions.put(runtime, java.lang.Iterable.class, (proxyClass) -> Iterable.define(runtime, proxyClass));
+        JavaExtensions.put(runtime, java.lang.Comparable.class, (proxyClass) -> Comparable.define(runtime, proxyClass));
+        JavaExtensions.put(runtime, java.lang.Throwable.class, (proxyClass) -> Throwable.define(runtime, (RubyClass) proxyClass));
+        JavaExtensions.put(runtime, java.lang.Runnable.class, (proxyClass) -> Runnable.define(runtime, proxyClass));
+        JavaExtensions.put(runtime, java.lang.Character.class, (proxyClass) -> Character.define(runtime, (RubyClass) proxyClass));
+        JavaExtensions.put(runtime, java.lang.Number.class, (proxyClass) -> Number.define(runtime, (RubyClass) proxyClass));
+        JavaExtensions.put(runtime, java.lang.Class.class, (proxyClass) -> Class.define(runtime, (RubyClass) proxyClass));
+        JavaExtensions.put(runtime, java.lang.ClassLoader.class, (proxyClass) -> ClassLoader.define(runtime, (RubyClass) proxyClass));
         // Java::byte[].class_eval ...
-        final RubyModule byteArray = Java.getProxyClass(runtime, new byte[0].getClass());
-        byteArray.addMethod("ubyte_get", new UByteGet(byteArray));
-        byteArray.addMethod("ubyte_set", new UByteSet(byteArray));
-
-        final RubyModule String = Java.getProxyClass(runtime, java.lang.String.class);
-        String.defineAlias("to_str", "to_s");
-
-        final RubyModule Number = Java.getProxyClass(runtime, java.lang.Number.class);
-        Number.defineAlias("to_int", "longValue");
-        Number.defineAlias("to_f", "doubleValue");
+        JavaExtensions.put(runtime, new byte[0].getClass(), (byteArray) -> {
+            byteArray.addMethod("ubyte_get", new UByteGet(byteArray));
+            byteArray.addMethod("ubyte_set", new UByteSet(byteArray));
+        });
+        JavaExtensions.put(runtime, java.lang.String.class, (proxyClass) -> {
+            proxyClass.defineAlias("to_str", "to_s");
+        });
     }
 
     @JRubyModule(name = "Java::JavaLang::Iterable", include = "Enumerable")
     public static class Iterable {
 
-        static RubyModule define(final Ruby runtime) {
-            final RubyModule Iterable = Java.getProxyClass(runtime, java.lang.Iterable.class);
+        static RubyModule define(final Ruby runtime, final RubyModule Iterable) {
```

Comment:
Could we use a different name for these? I find it extremely confusing when I look at the code and I see "Iterable" meaning two different things.

Reply:
Note that at this point in the code, Iterable now means three different things: the inner class, the variable, and (to me at a glance) the actual Java Iterable class. It's quite confusing.

Reply:
Same comment applies to all other such names below. I understand that it might look a bit nicer to some, but this flies against typical Java local variable naming convention.

It's somewhat confusing that the extension classes have the same names as the Java class they extend, but that seems like less of an issue to me than the local variables having yet again the same name.

Reply:
okay, will take a look into that